pytorch_lightning/trainer/lightning_config.py

```diff
@@ -0,0 +1,247 @@
+from typing import Union, Optional, List, Dict, Iterable
+from dataclasses import dataclass, fields
+
+from pytorch_lightning.loggers import LightningLoggerBase
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping, Callback
+from pytorch_lightning.profiler import BaseProfiler
+
+
+@dataclass
+class LightningConfig:
```

Comment:
this may have a disadvantage that all the arguments won't be listed in `Trainer` docs page and so a user needs to search for it also it drops IDE support of whispering possible arguments